criu/cr-restore.c

```diff
@@ -3824,15 +3824,19 @@ static int sigreturn_restore(pid_t pid, struct task_restore_args *task_args, uns
 
 	__gcov_flush();
 
-	pr_info("task_args: %p\n"
+	/*This message is commented out as
+	it causes a segmentation fault when restore
+	is run while using binary logging*/
```

Comment:
Have you managed to find out why it crashes?

Reply:
I did some investigation on this and found that it is linked to the default value of -1 assigned to the flog_genbit in the _Generic() macro. Today, when I changed it to 0, and the restore ran successfully with these messages as well. Should I make the _Generic return 0 as the default value to avoid this crash?

Reply:
No, please, find out why -1 is reported. Maybe it's unsupported data type and/or format specified?

Reply:
The argument that is trying to be printed is a pointer of type _struct task_restore_args *_ and, this doesn’t fit into any of the other specified data types in the _Generic macro, and hence the -1 is returned. 
 
If I include this struct type explicitly in the _Generic list, the error is resolved. But in this pr_info, _task_args->clone_restore_fn_, is a function pointer. I had typecasted it to (void *) and then it did not return an error.